**OBJECTIVE** :
*Analyze employee attrition and identify reasons patterns for employee exits, including department, job role, overtime, promotions, tenure, compensation.*

***BUSINESS GOAL*** : Provide HR stakeholders with data- driven insights into potential employee attrition patterns and data- quality issues.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


In [2]:
raw_df = pd.read_csv('Employee_stats.project3.csv')
df = raw_df.copy()
df

,Employee_ID,Full_Name,Gender,Department,Job_Level,Job_Role,Location,Joining_Date,Exit_Date,Employment_Status,...,Overtime,Manager_ID,Years_At_Company,Training_Hours,Promotions,Absences,Recruitment_Source,Remote_Work_Days,Attrition,Exit_Reason
0,EMP105115,Employee_5114,Female,Engineering,L1,QA Engineer,Delhi NCR,2017-11-30,NaN,Active,...,No,MGR1248,11.3,16.5,4,7,Agency,1,No,NaN
1,EMP101385,Employee_1384,Female,Operations,L2,Marketing Analyst,Delhi NCR,2021-01-31,NaN,Active,...,No,MGR1195,8.7,8.7,0,1,Agency,5,No,NaN
2,EMP101646,Employee_1645,Female,Sales,L1,HR Executive,Mumbai,2017-04-15,NaN,Active,...,Yes,MGR1098,8.6,12.2,2,3,Company Website,5,No,NaN
3,EMP100304,Employee_0303,Male,Marketing,L4,Sales Executive,Chennai,2023-09-12,NaN,Active,...,Yes,MGR1181,10.4,9.6,1,9,LinkedIn,4,No,NaN
4,EMP102393,Employee_2392,Female,Operations,L2,Sales Executive,Bengaluru,2025-05-20,NaN,Active,...,No,MGR1167,2.8,16.7,1,4,Referral,0,No,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6560,EMP106478,Employee_6477,Male,Operations,L2,QA Engineer,Pune,2020-06-07,NaN,Active,...,Yes,MGR1187,6.4,10.8,2,5,Naukri,5,No,NaN
6561,EMP102126,Employee_2125,Male,Customer Support,L5,HR Analyst,Chennai,2022-11-19,2028-09-04,Exited,...,No,MGR1142,6.0,5.1,0,5,Agency,1,Yes,NaN
6562,EMP105049,Employee_5048,Male,Operations,L4,Marketing Analyst,Bengaluru,2024-01-15,NaN,Active,...,Yes,MGR1071,5.2,23.4,0,5,LinkedIn,2,No,NaN
6563,EMP100795,Employee_0794,Male,Operations,L1,Support Executive,Bengaluru,2019-08-17,NaN,Active,...,No,MGR1085,4.7,7.4,1,4,Referral,5,No,NaN


In [3]:
df.head()
df.shape
df.info()
df.describe(include = "all")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6565 entries, 0 to 6564
Data columns (total 25 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Employee_ID          6565 non-null   object 
 1   Full_Name            6565 non-null   object 
 2   Gender               6565 non-null   object 
 3   Department           6565 non-null   object 
 4   Job_Level            6565 non-null   object 
 5   Job_Role             6466 non-null   object 
 6   Location             6565 non-null   object 
 7   Joining_Date         6547 non-null   object 
 8   Exit_Date            1382 non-null   object 
 9   Employment_Status    6565 non-null   object 
 10  Age                  6483 non-null   float64
 11  Monthly_Salary       6456 non-null   float64
 12  Performance_Score    6485 non-null   float64
 13  Engagement_Score     6444 non-null   float64
 14  Work_Hours_Per_Week  6565 non-null   float64
 15  Overtime             6565 non-null   o

,Employee_ID,Full_Name,Gender,Department,Job_Level,Job_Role,Location,Joining_Date,Exit_Date,Employment_Status,...,Overtime,Manager_ID,Years_At_Company,Training_Hours,Promotions,Absences,Recruitment_Source,Remote_Work_Days,Attrition,Exit_Reason
count,6565,6565,6565,6565,6565,6466,6565,6547,1382,6565,...,6565,6484,6565.000000,6565.000000,6565.000000,6565.000000,6465,6565.000000,6565,1350
unique,6500,6530,9,15,5,12,6,3001,1162,2,...,10,249,NaN,NaN,NaN,NaN,6,NaN,8,7
top,EMP102438,Employee_5461,Male,Sales,L2,Support Executive,Bengaluru,2017-10-23,2025-06-15,Active,...,No,MGR1009,NaN,NaN,NaN,NaN,Agency,NaN,No,Better Opportunity
freq,2,2,3487,875,1935,587,1154,9,25,5188,...,4443,38,NaN,NaN,NaN,NaN,1119,NaN,5128,204
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,5.959238,15.698568,0.805179,5.018736,NaN,2.485453,NaN,NaN
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,3.269665,10.537295,1.011649,3.185861,NaN,1.696497,NaN,NaN
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,0.300000,0.200000,-3.000000,-5.000000,NaN,0.000000,NaN,NaN
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,3.100000,8.000000,0.000000,3.000000,NaN,1.000000,NaN,NaN
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,6.000000,13.300000,1.000000,5.000000,NaN,3.000000,NaN,NaN
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,8.700000,21.200000,1.000000,6.000000,NaN,4.000000,NaN,NaN


In [4]:
df.isna().sum()
df.nunique()

,0
Employee_ID,6500
Full_Name,6530
Gender,9
Department,15
Job_Level,5
Job_Role,12
Location,6
Joining_Date,3001
Exit_Date,1162
Employment_Status,2


In [5]:
df.duplicated().sum()

np.int64(35)

**Data quality issue** : The datatset contains numerous missing values over several columns, inconsistent values, questionable demographic values.

In [6]:
df["Attrition"].value_counts(normalize = True)*100

,proportion
Attrition,
No,78.111196
Yes,20.045697
YES,0.319878
N,0.304646
NO,0.304646
Y,0.304646
no,0.304646
yes,0.304646


In [7]:
df["Attrition"].value_counts(dropna=False)

,count
Attrition,
No,5128
Yes,1316
YES,21
N,20
NO,20
Y,20
no,20
yes,20


In [8]:
df["Attrition"] = df["Attrition"].str.strip().str.title()

In [9]:
df["Attrition"] = df["Attrition"].replace({
   "Y" : "Yes",
  "N": "No"
})

In [10]:
df["Attrition"].value_counts(dropna=False)

,count
Attrition,
No,5188
Yes,1377


In [11]:
df["Department"].value_counts(dropna=False)

,count
Department,
Sales,875
Operations,860
Engineering,845
Marketing,807
Finance,784
Customer Support,756
IT,753
HR,705
IT,30


In [12]:
df["Department"] = df["Department"].str.strip().str.title()

In [13]:
df["Department"] = df["Department"].replace ({
    "Human Resources" : "HR",
    "Hr" : "HR",
    "Cust Support" : "Customer Support",
    "It" : "IT"
})

In [14]:
df["Department"].value_counts(dropna=False)

,count
Department,
Sales,875
Operations,860
Engineering,845
Customer Support,816
IT,813
Marketing,807
Finance,784
HR,765


Department column was found with inconsistent capitilization and spacing problems. these problems were fixed so one department isnt considered more times than one.

In [15]:
df["Gender"] = df["Gender"].str.strip().str.title()

In [16]:
df["Gender"] = df["Gender"].replace({
    "M" : "Male",
    "F" : "Female"
})

In [17]:
df["Gender"].value_counts(dropna=False)

,count
Gender,
Male,3547
Female,2899
Non-Binary,119


In [18]:
df["Joining_Date"] = pd.to_datetime(df["Joining_Date"],
                                    errors = "coerce")

In [19]:
df["Exit_Date"] = pd.to_datetime(df["Exit_Date"],
                                    errors = "coerce")

invalid/missing dates became naT

In [20]:
df[df["Joining_Date"].dt.year > 2026][["Employee_ID", "Joining_Date", "Exit_Date", "Attrition"]]

,Employee_ID,Joining_Date,Exit_Date,Attrition


In [21]:
 df[df["Exit_Date"].dt.year > 2026][["Employee_ID", "Joining_Date", "Exit_Date", "Attrition"]]

,Employee_ID,Joining_Date,Exit_Date,Attrition
31,EMP102330,2023-12-28,2030-12-07,Yes
42,EMP101173,2021-11-26,2027-07-09,Yes
80,EMP101168,2021-09-03,2029-01-16,Yes
93,EMP102434,2025-12-12,2028-08-04,Yes
101,EMP106336,2025-06-08,2027-03-20,Yes
...,...,...,...,...
6536,EMP103120,2025-01-15,2028-03-31,Yes
6538,EMP100202,2023-03-10,2027-05-16,Yes
6541,EMP101853,2022-08-16,2029-04-28,Yes
6554,EMP104535,2025-03-17,2032-08-27,Yes


In [22]:
future_exits = df[df["Exit_Date"].dt.year > 2026].copy()

future_exits[
["Employee_ID", "Joining_Date", "Exit_Date", "Attrition"]
].head(20)


,Employee_ID,Joining_Date,Exit_Date,Attrition
31,EMP102330,2023-12-28,2030-12-07,Yes
42,EMP101173,2021-11-26,2027-07-09,Yes
80,EMP101168,2021-09-03,2029-01-16,Yes
93,EMP102434,2025-12-12,2028-08-04,Yes
101,EMP106336,2025-06-08,2027-03-20,Yes
134,EMP104523,2024-11-08,2029-03-22,Yes
137,EMP105732,2025-06-05,2033-02-09,Yes
164,EMP105991,2022-06-05,2028-10-30,Yes
186,EMP102628,2024-10-23,2031-10-19,Yes
202,EMP101017,2021-11-15,2029-12-04,Yes


In [23]:
future_exits["Attrition"].value_counts(dropna=False)


,count
Attrition,
Yes,482


In [24]:

future_exits["Exit_Date"].dt.year.value_counts().sort_index()

,count
Exit_Date,
2027,128
2028,103
2029,99
2030,64
2031,47
2032,23
2033,18


In [25]:
df.loc[df["Exit_Date"].dt.year > 2026, "Exit_Date"] = pd.NaT

In [26]:
df["Tenure_Years"] = (
    (df["Exit_Date"] - df["Joining_Date"]).dt.days/365.25).round(1)

In [27]:
Tenure_band = pd.cut(
    df["Tenure_Years"],
    bins = [ 0, 1, 3, 5, float("inf")],
    labels = [ "<1year", "1-3years", "3-5years", "5=years"]
)

In [28]:
df["Joining_Date"].isna().sum()

np.int64(27)

In [29]:
df["Exit_Date"].isna().sum()

np.int64(5665)

In [30]:
print("Future exit dates remaining:",
(df["Exit_Date"].dt.year > 2026).sum())



Future exit dates remaining: 0


In [31]:
print("Missing exit dates:",
df["Exit_Date"].isna().sum())



Missing exit dates: 5665


In [32]:
df.groupby("Attrition")["Exit_Date"].apply(lambda x: x.isna().sum())


,Exit_Date
Attrition,
No,5163
Yes,502


In [33]:
df["Attrition"].value_counts(dropna=False)

,count
Attrition,
No,5188
Yes,1377


In [34]:
df[
    (df["Age"] < 18) |
    (df["Age"] > 65)
][["Employee_ID", "Age"]]

,Employee_ID,Age
70,EMP105887,17.0
988,EMP102663,14.0
1154,EMP105347,75.0
1467,EMP102365,99.0
1500,EMP101125,16.0
2523,EMP105186,82.0
2651,EMP102816,0.0
2809,EMP100594,73.0
3567,EMP104954,71.0
3615,EMP102311,120.0


Demographic values out of legal or usual working age constraints were found.

In [35]:
df["Overtime"] = df["Overtime"].str.strip().str.title()

In [36]:
df["Overtime"].value_counts(dropna=False)

,count
Overtime,
No,4488
Yes,2047
N,15
Y,15


In [37]:
df["Overtime"] = df["Overtime"].replace({
    "N" : "No",
    "Y" : "Yes"
})

In [38]:
df["Overtime"].value_counts(dropna=False)

,count
Overtime,
No,4503
Yes,2062


Overtime column also was found with inconsistent naming, spacing and capitilization issues.

In [39]:
exited = df[df["Attrition"] == "Yes"]

In [40]:
df.loc[df["Promotions"] < 0, "Promotions"]= np.nan

In [41]:
df["Promotions"].value_counts(dropna = False)

,count
Promotions,
0.0,2938
1.0,2412
2.0,894
3.0,251
4.0,47
5.0,16
NaN,3
30.0,1
20.0,1


In [42]:
df.groupby("Attrition")["Work_Hours_Per_Week"].mean()

,Work_Hours_Per_Week
Attrition,
No,42.947398
Yes,42.701017


Training hours, work hours per week and monthly salary largely remains the same among existing and exited employees

**Q1) What are the main reasons employees are leaving?**

In [43]:
pd.crosstab(df["Attrition"],df["Exit_Reason"])

Exit_Reason,Better Opportunity,Career Growth,Compensation,Manager,Personal Reasons,Relocation,Workload
Attrition,,,,,,,
No,26,0,0,0,0,0,0
Yes,178,202,186,204,180,187,187


In [44]:
df[df["Attrition"]=="Yes"]["Exit_Reason"].value_counts()

,count
Exit_Reason,
Manager,204
Career Growth,202
Workload,187
Relocation,187
Compensation,186
Personal Reasons,180
Better Opportunity,178


In [45]:
manager_exits = df[
   (df ["Attrition"] == "Yes") &
   (df ["Exit_Reason"] == "Manager")
]

In [46]:
manager_exits["Manager_ID"].value_counts()

,count
Manager_ID,
MGR1239,5
MGR1180,4
MGR1011,4
MGR1088,4
MGR1129,3
...,...
MGR1128,1
MGR1023,1
MGR1185,1


**Q2) Are compensation exits concentrated among certain groups?**

In [47]:
comp_exits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Compensation")
    ]

In [48]:
comp_exits["Monthly_Salary"].describe()

,Monthly_Salary
count,182.000000
mean,65907.142857
std,23303.595710
min,22000.000000
25%,48925.000000
50%,67650.000000
75%,80250.000000
max,134500.000000


In [49]:
comp_exits["Department"].value_counts()

,count
Department,
Engineering,31
Operations,30
Customer Support,24
Finance,24
IT,22
Sales,20
HR,18
Marketing,17


In [50]:
comp_exits["Job_Role"].value_counts()

,count
Job_Role,
HR Executive,25
HR Analyst,20
Data Analyst,19
Recruiter,18
Sales Executive,16
Account Manager,16
Support Executive,15
Software Engineer,15
Financial Analyst,14


Compensation exits concentrated among engineering and operations department.

In [51]:
High_salary_compexits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Compensation") &
    (df["Monthly_Salary"] > 30000) &
    (df["Tenure_Years"] < 3) &
    (df["Overtime"] == "No") &
    (df["Promotions"] > 0)
]

Employees with less than 3 years at the company, no overtime and with promotions more ,than 30000  as salary  still listed compensation as exit reasons.

In [52]:
High_salary_compexits["Department"].value_counts()


,count
Department,
Operations,7
IT,3
Finance,2
Sales,2
Engineering,2
HR,1
Customer Support,1


In [53]:
low_salary_compexits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Compensation") &
    (df["Monthly_Salary"] < 30000) &
    (df["Tenure_Years"] > 3) &
    (df["Overtime"] == "Yes") &
    (df["Promotions"] == 0)
]

In [54]:
low_salary_compexits["Department"].value_counts()

,count
Department,
HR,1


Employees with more than 3 years tenure at the company, no promotions, overtime and less than 30000 as salary listed compensation as exit reason, but only 2 employees as such were mentioned exited.

**Q3) Did workload exits in anyway related to them working overtime?**

In [55]:
Workload_exits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Workload")
]

In [56]:
pd.crosstab(Workload_exits["Overtime"], Workload_exits["Department"])

Department,Customer Support,Engineering,Finance,HR,IT,Marketing,Operations,Sales
Overtime,,,,,,,,
No,12,21,14,18,20,17,21,21
Yes,2,4,4,6,6,5,8,8


In [57]:
pd.crosstab(Workload_exits["Work_Hours_Per_Week"], Workload_exits["Overtime"])

Overtime,No,Yes
Work_Hours_Per_Week,,
12.0,1,0
28.9,1,0
29.1,1,0
30.3,0,1
31.8,0,1
...,...,...
52.1,3,0
52.2,1,0
52.4,1,0


In [58]:
Workload_exit = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Workload") &
    (df["Work_Hours_Per_Week"] > 35) &
    (df["Overtime"] == "Yes")
]

In [59]:
Workload_exit["Work_Hours_Per_Week"].value_counts()

,count
Work_Hours_Per_Week,
40.0,2
44.7,2
35.4,1
47.4,1
48.5,1
42.0,1
38.9,1
44.4,1
45.0,1


In [60]:
Betteropp_exits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Better_Opportunity")
]

In [61]:
Career_growth_exits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Career_Growth")
]


In [62]:
Personalreasons_exits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "PersonalReasons")
]

In [63]:
Relocation_exits = df[
    (df["Attrition"] == "Yes") &
    (df["Exit_Reason"] == "Relocation")
]

**Q4) Is overtime associated with higher attrition?**

In [64]:
pd.crosstab(
    df["Overtime"],
    df["Attrition"],
    normalize="index"
) * 100

Attrition,No,Yes
Overtime,,
No,79.213857,20.786143
Yes,78.612997,21.387003


**Q5) What is the overall Attrition rate?**

In [65]:
df["Attrition"].value_counts(normalize = True)*100

,proportion
Attrition,
No,79.025133
Yes,20.974867


**Q6) Are promotions anyway associated with the attrition rate? among exited employees, how many promotions did thry have?**

In [66]:
pd.crosstab(
    df["Promotions"],
    df["Attrition"])

Attrition,No,Yes
Promotions,,
0.0,2309,629
1.0,1904,508
2.0,708,186
3.0,210,41
4.0,38,9
5.0,14,2
6.0,1,0
15.0,0,1
20.0,0,1


In [67]:
pd.crosstab(
  df["Promotions"],
  df["Attrition"],
  normalize = "index"
  )* 100

Attrition,No,Yes
Promotions,,
0.0,78.590878,21.409122
1.0,78.938640,21.061360
2.0,79.194631,20.805369
3.0,83.665339,16.334661
4.0,80.851064,19.148936
5.0,87.500000,12.500000
6.0,100.000000,0.000000
15.0,0.000000,100.000000
20.0,0.000000,100.000000


In [68]:
pd.crosstab(exited["Promotions"],exited["Attrition"])

Attrition,Yes
Promotions,
0,629
1,508
2,186
3,41
4,9
5,2
15,1
20,1


**Q7) Are employee exits related to their engagement score or performance score? Do they differ between existing and exited employees?**

In [69]:
pd.crosstab(df["Attrition"],df["Engagement_Score"])

Engagement_Score,-1.0,0.0,1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0,10.0
Attrition,,,,,,,,,,,,
No,1,2,237,747,2030,1544,525,1,1,1,1,1
Yes,0,0,54,199,526,425,149,0,0,0,0,0


In [70]:
df.groupby("Attrition")["Engagement_Score"].mean()

,Engagement_Score
Attrition,
No,3.272638
Yes,3.307465


In [71]:
pd.crosstab(df["Attrition"],df["Performance_Score"])

Performance_Score,-1.0,0.0,1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0,10.0
Attrition,,,,,,,,,,,,
No,1,2,200,786,2195,1519,420,1,1,1,1,1
Yes,0,0,63,232,571,392,99,0,0,0,0,0


In [72]:
df.groupby("Attrition")["Performance_Score"].mean()

,Performance_Score
Attrition,
No,3.231669
Yes,3.170965


**Q8) Is there any connection between Attrition and department? what is the Attrition rate for each department?**

In [73]:
pd.crosstab(df["Attrition"],df["Department"])

Department,Customer Support,Engineering,Finance,HR,IT,Marketing,Operations,Sales
Attrition,,,,,,,,
No,661,660,607,615,643,634,673,695
Yes,155,185,177,150,170,173,187,180


In [74]:
pd.crosstab(
    df["Department"],
    df["Attrition"],
    normalize = "index")["Yes"].sort_values(ascending = False)*100

,Yes
Department,
Finance,22.576531
Engineering,21.893491
Operations,21.744186
Marketing,21.437423
IT,20.910209
Sales,20.571429
HR,19.607843
Customer Support,18.995098


**Q9) What is the Attrition rate among different genders?**

In [75]:
pd.crosstab(df["Gender"], df["Attrition"],
            normalize =  "index"
            ) * 100

Attrition,No,Yes
Gender,,
Female,78.509831,21.490169
Male,79.221878,20.778122
Non-Binary,85.714286,14.285714


In [76]:
pd.crosstab(
    exited["Gender"],
    exited["Attrition"],
)

Attrition,Yes
Gender,
Female,623
Male,737
Non-Binary,17


**Q10)How do training hours differ between exited and active employees?**

In [77]:
df.groupby("Attrition")["Training_Hours"].mean()

,Training_Hours
Attrition,
No,15.652776
Yes,15.871097


**Q11) Which Job roles have the highest attrition rate?**

In [78]:
pd.crosstab(
    df["Job_Role"],
    df["Attrition"],
    normalize = "index"
    ) * 100

Attrition,No,Yes
Job_Role,,
Account Manager,79.444444,20.555556
Data Analyst,76.311031,23.688969
Financial Analyst,81.108830,18.891170
HR Analyst,77.325581,22.674419
HR Executive,76.739927,23.260073
Marketing Analyst,78.890877,21.109123
Operations Analyst,81.010453,18.989547
QA Engineer,78.957169,21.042831
Recruiter,77.929688,22.070312


**Q12) Is salary associated with attrition? How does salary differ among active and exited employees?**

In [79]:
df.groupby("Attrition")["Monthly_Salary"].mean()

,Monthly_Salary
Attrition,
No,65817.156667
Yes,65320.353982


In [80]:
df.groupby("Attrition")["Monthly_Salary"].agg(["count", "mean", "median"])

,count,mean,median
Attrition,,,
No,5100,65817.156667,65050.0
Yes,1356,65320.353982,65000.0


# **KEY FINDINGS**


1. **Manager** : Manager- related issues and career growth were the most frequently reported exit reasons, followed by workload, relocation, compensation, personal reasons and better oppurtunities.

2. **Compensation** : Compensation-related exits occured across multiple departments and salary levels, Some employees with less than 3 years of tenure at the company,salaries above 30,000 and atleast one promotion still cited compensation as exit reason.

3. **Workload** : Workload-related exits occured among both employees with and without overtime. A subset of employees citing workload also worked more than 35 hours per week along with overtime.

4. **Performance and Engagement** : Performance score and Engagement score among active and exited employees largely remains the same, no substantial difference was observed in the dataset.

5. **Overtime** : Attrition was 20.7% among employees working no overtime and 21.3% among employees who worked overtime.

6. **Gender** : Female employees had an attrition rate of 21.5%, compared with 20.8% among male employees and 14.3% among non-binary employees.

7. **Department** : Finance had the highest department-level attrition rate at approximately 22.6%, while customer support had the lowest at approximately 18.99%.

8. **Promotions** : Attrition was highest among employees with 0-2 promotions, ranging from approximately 20.8% to 21.4%. Attrition was lower among employees with 3 or more promotions, although some promotion group contain extreme values such as 15, 20 and 30 , these require data- quality reviews.

8. **Salary and working conditions** : Average monthly salary, work hours per week, training hours remain broadly similar for exited and active employees.





In [81]:
df.to_csv("Employee_stats_project3_cleaned.csv", index=False)